sdcm/teardown_validators/events.py

```diff
@@ -0,0 +1,27 @@
+import logging
+import re
+
+from sdcm.sct_events import Severity
+from sdcm.sct_events.file_logger import get_events_grouped_by_category
+from sdcm.teardown_validators.base import TeardownValidator
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ErrorEventsValidator(TeardownValidator):  # pylint: disable=too-few-public-methods
+    validator_name = 'error_events'
+
+    def validate(self):
+        # Do not run validator if there were no parallel nemeses
+        if len(self.tester.get_nemesis_class()) == 1:
+            return
+
+        LOGGER.info("Running validation of ERROR events for parallel nemeses")
+        error_events = get_events_grouped_by_category(
+            _registry=self.tester.events_processes_registry)[Severity.ERROR.name]
+        failing_event_types = self.configuration.get("failing_events")
+        pattern = re.compile('|'.join(failing_event_types), re.IGNORECASE)
+        failing_events = [event for event in error_events if pattern.search(event)]
```

Comment:
I'd propose to iterate over `get_events_main_device(_registry=self.tester.events_processes_registry).raw_events_log` (I'm not sure of this line) and parse each line (the're jsons) and match by e.g. event type/subtype. It will give us more possibilities as pattern search is limited and error prone.

Reply:
Good suggestion, thank you
After changing to the suggested approach it is much better to work with native python data structure